Tests/TuistKitTests/Storyboards/Mocks/MockStoryboardGenerator.swift

```diff
@@ -0,0 +1,33 @@
+import Basic
+import Foundation
+@testable import TuistKit
+
+final class MockStoryboardGenerator: StoryboardGenerating {
+    typealias Storyboard = (path: AbsolutePath, name: String, platform: Platform, product: Product, isLaunchScreen: Bool)
+
+    var generatedStoryboards: [Storyboard] = []
+    var generateStub: Error?
+
+    func generate(path: AbsolutePath, name: String, platform: Platform, product: Product, isLaunchScreen: Bool) throws {
+        let storyboard = Storyboard(path, name, platform, product, isLaunchScreen)
+
```

Comment:
Seeing this is a mock, we can give it minimal logic, it can simply append storyboards without it internally doing any verifications, in the test we verify `generatedStoryboards` matches our expectations

To simplify the comparisons, we could convert `Storyboard` > `GeneratedStoryboard` and make it `Equatable` - that way in our test the storyboard section could be condensed to:

```swift
XCTAssertEqual(storyboardGenerator.generatedStoryboards, [
   GeneratedStoryboard(path: sourcesPath, name: "Launch Screen", platform: .iOS, product: .app, isLaunchScreen: true),
   GeneratedStoryboard(path: sourcesPath, name: "Main", platform: .iOS, product: .app, isLaunchScreen: false)
])
```

Reply:
I’ll give this a go. Thanks!

Reply:
👍 - just realized I missed "struct" in my suggestion above